File: src/main/resources/static/js/music/musicNavigation.js
// Music View
var MusicView = (function () {

  var musicPanelListElement

  var musicPanelInfoTemplateElement;

  $(document).ready(function(){
    musicPanelListElement = $('#music-navbar-content').find('#music-panel-list');
    musicPanelInfoTemplateElement = $('#music-panel-info');
    musicPanelInfoTemplateElement.show();
  })

  var fromUsername = function(actionToDo){
    var currentUser = CurrentUser.retrieveCurrentUser()
    actionToDo(currentUser.username)
  }

  var refreshMusicListPanel = function () {
    fromUsername(function(username){
      $.ajax({
        method: 'GET',
        url: 'user/' + username + '/music/all',
        success: function (musicManagerType) {
          musicPanelInfoTemplateElement.show()
          var numberOfMusic = musicManagerType.musicTypeDTO.length

          if(numberOfMusic != 0) {
            musicPanelListElement.empty()
            var listMusicType = musicManagerType.musicTypeDTO
            for (i = 0; i < listMusicType.length; i++) {
              var musicPanelInfo = musicPanelInfoTemplateElement.clone().removeClass('out-of-bound')
              musicPanelInfo.find('#music-iframe').attr('src', listMusicType[i].sourceUrl)
              musicPanelInfo.show()
              musicPanelListElement.append(musicPanelInfo)
            }
            musicPanelListElement.show()
          } else {
            AddMusicModal.showModal();
            //you have no music !
          }

        }
      })
    })
  }

  return {
    displayMusicPanel: function () {
      musicPanelListElement.show()
      AddMusicModal.hideModal()
      refreshMusicListPanel()
    }
  }
})()

File: src/scala/com/ldz/music/navigation/MusicNavigation.scala
package com.ldz.music.navigation

import com.ldz.addMusicModal.AddMusicPopup
import com.ldz.javascript.face.CurrentUser
import com.ldz.music.navigation.jquery.RefreshMusicJQuery
import org.scalajs.dom.document
import org.scalajs.jquery.{JQuery, jQuery}

import scala.scalajs.js.annotation.{JSExport, JSExportTopLevel}

/**
  * Created by Loic on 25/06/2017.
  */
@JSExportTopLevel("MusicView")
object MusicNavigation {

  var musicPanelListElement: JQuery = null
  var musicPanelInfoTemplateElement: JQuery = null
  var addMusicButton: JQuery = null

  jQuery(document).ready(() => {
    musicPanelListElement = jQuery("#music-navbar-content").find("#music-panel-list")
    musicPanelInfoTemplateElement = jQuery("#music-panel-info")
    musicPanelInfoTemplateElement.show()
    addMusicButton = jQuery("#add-music-button")
    addMusicButton.click(() => {
      AddMusicPopup.showModal()
    })
  })

  private def refreshMusicListPanel(): Unit = {
    val username = CurrentUser.retrieveCurrentUser().username
    jQuery.ajax(s"user/${username}/music/all", RefreshMusicJQuery.apply(username))
  }

  @JSExport
  def displayMusicPanel(): Unit = {
    musicPanelListElement.show()
    AddMusicPopup.hideModal()
    refreshMusicListPanel()
  }

}